### PR TITLE
refactor: move users and workspaces schemas under schemas module

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR moves user and workspace schemas defined in the `security` module to the proper `schemas.v0` module.

Also, the `users_file` setting attribute is removed, but the corresponding environment variable is still available from the `user migrate` command. This change does not change the current behaviour.

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

Running tests locally

**Checklist**

- [ ] I added relevant documentation
- [X] I followed the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/server/schemas/v0/users.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+USER_PASSWORD_MAX_LENGTH = 100
+
+
+class UserCreate(BaseModel):
+    first_name: constr(min_length=1, strip_whitespace=True)
+    last_name: Optional[constr(min_length=1, strip_whitespace=True)]
+    username: str = Field(regex=USER_USERNAME_REGEX, min_length=1)
+    role: Optional[UserRole]
+    password: constr(min_length=USER_PASSWORD_MIN_LENGTH, max_length=USER_PASSWORD_MAX_LENGTH)
+    workspaces: Optional[List[str]]
```
Comment (jfcalvo) [inline src/argilla/server/schemas/v0/users.py]: In the refactor that I did for API v1 schemas I'm trying to define first when possible the canonical entity (in this case the `User`) and later schemas that create/modify (like `UserCreate` in this case). So my suggestion is moving this schema class to the end of the file.

File: src/argilla/server/schemas/v0/workspaces.py [hunk 2]
```diff
@@ -12,11 +12,29 @@
...
-def validate_is_super_user(user: User, message: str = None):
-    """Common validation to ensure the current user is a admin/superuser"""
-    if not user.is_superuser():
-        raise ForbiddenOperationError(message or "Only admin users can apply this change")
+WORKSPACE_NAME_REGEX = ES_INDEX_REGEX_PATTERN
+
+
+class WorkspaceUserCreate(BaseModel):
+    user_id: UUID
+    workspace_id: UUID
```
Comment (jfcalvo) [inline src/argilla/server/schemas/v0/workspaces.py]: The same here, maybe you can move this class to the end of the file.

**Review by jfcalvo — APPROVED** [review]
Looks good to me. I added two small possible refactors in the case that you want to apply them.

File: src/argilla/server/schemas/v0/workspaces.py (hunk 2 above)
Comment (frascuchon) [reply src/argilla/server/schemas/v0/workspaces.py]: done. Thanks

**Pull request merged**